api/cmd/etcd-launcher/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const (
+	initialStateEnvName      = "INITIAL_STATE"
+	initialClusterEnvName    = "INITIAL_CLUSTER"
+	defaultClusterSize       = 3
+	defaultEtcdctlAPIVersion = "3"
+)
+
+type envConfig struct {
+	namespace             string
+	clusterSize           int
+	podName               string
+	podIP                 string
+	etcdctlAPIVersion     string
+	dataDir               string
+	token                 string
+	enableCorruptionCheck bool
+}
+
+func main() {
+	// normal workflow, we don't do migrations anymore
+	config, err := getConfigFromEnv()
+	if err != nil {
+		log.Fatalf("failed to get launcher configuration: %v", err)
+	}
+
+	// not required, will leave it for now.
+	os.Setenv(initialStateEnvName, "new")
+	os.Setenv(initialClusterEnvName, initialMemberList(config.clusterSize, config.namespace))
+
+	log.Print("initializing etcd..")
+	log.Printf("initial-state: %s", os.Getenv(initialStateEnvName))
+	log.Printf("initial-cluster: %s", os.Getenv(initialClusterEnvName))
+
+	err = syscall.Exec(
+		"/usr/local/bin/etcd",
+		etcdCmd(config),
+		os.Environ())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+}
+
+func initialMemberList(n int, namespace string) string {
+	members := []string{}
+	for i := 0; i < n; i++ {
+		members = append(members, fmt.Sprintf("etcd-%d=http://etcd-%d.etcd.%s.svc.cluster.local:2380", i, i, namespace))
+	}
+	return strings.Join(members, ",")
+}
+
+func getConfigFromEnv() (*envConfig, error) {
+	config := &envConfig{}
+	if n := os.Getenv("NAMESPACE"); n != "" {
```

Comment:
this pattern (here and in code below) `if some := os.Getenv("NAMESPACE"); some != ""` could be improved with `os.LookupEnv()`

```
var ok bool
config.namespace, ok = os.LookupEnv("NAMESPACE")
if !ok {
... error out
}
```

the point is, it doesn't require `else` blocks.